### PR TITLE
docs: Add project documentation

### PR DESCRIPTION
**Pull request by elias-gvin**
1. Added instruction on environment setup
2. Added document with code conventions:
such as code style and commit message convention

### CODE REVIEW

**Review by SurprisedBird — DISMISSED** [review]

**Comment by elias-gvin** [comment]
> It's unclear why we should to avoid double underscore prefix.

Just for the sake of consistency: so every protected field or method in our code will use the same mechanism of name hiding (and will look the same)
I believe that "private" fields (python, actually have not private fields and "double underscore" mechanism is a dirty hack, actually) are not really needed in our project.
But still - it is only a recommendation. If we'll face situation when "private" field is really needed - we can use it.

**Comment by ADetilie** [comment]
> Not clear what setter property means. Its better to use "property setter".

Ok

File: docs/CODE_CONVENTIONS.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        self._field_2 = create_field_method()
+        # Field type IS hinted because it is no way to
+        # understand which elements should contain this list - during initialization
+        self._field_3: List[SomeType3] = []
+    
+    # All method arguments marked with type hints
+    # Argument2Type is written inside quotes to avoid unnecessary import of Argument2Type in current module
+    def some_method_1(self, argument_1: Argument1Type, argument_2: 'Argument2Type') -> ReturnValueType:
+        # Local variable type should not be hinted
+        local_varuable = LocalVariable()
```
Comment (asternotus) [inline docs/CODE_CONVENTIONS.md]: Mistake. "local_variable" name should be

**Review by asternotus — CHANGES_REQUESTED** [review]
Please change it o give an answer.

File: docs/CODE_CONVENTIONS.md (hunk 1 above)
Comment (elias-gvin) [reply docs/CODE_CONVENTIONS.md]: Done

**Review by SurprisedBird — APPROVED** [review]

**Review by asternotus — APPROVED** [review]
Well done!

**Pull request merged**